src/game_interpreter_battle.cpp

```diff
@@ -597,12 +597,84 @@ bool Game_Interpreter_Battle::CommandManiacControlBattle(lcf::rpg::EventCommand
 	return true;
 }
 
-bool Game_Interpreter_Battle::CommandManiacControlAtbGauge(lcf::rpg::EventCommand const&) {
+bool Game_Interpreter_Battle::CommandManiacControlAtbGauge(lcf::rpg::EventCommand const& com) {
 	if (!Player::IsPatchManiac()) {
 		return true;
 	}
 
-	Output::Warning("Maniac Patch: Command ControlAtbGauge not supported");
+	int target_flags = com.parameters[0];
+	int target_reference_flags = com.parameters[1];
+	int target_reference_identifier = com.parameters[2];
+	int operation_flags = com.parameters[3];
+	int operand_flags = com.parameters[4];
+	int value_reference_flags = com.parameters[5];
+	int value_reference_identifier = com.parameters[6];
+
+	int target_id = ValueOrVariable(target_reference_flags, target_reference_identifier);
+
+	auto getAtbValue = [this, value_reference_flags, value_reference_identifier](int flags) {
+		int value = ValueOrVariable(value_reference_flags, value_reference_identifier);
+
+		switch (flags) {
+			case 0:
+				// value
+				return value;
+				break;
+			case 1:
+				// percentage
+				return (int) ((double)value / 100 * Game_Battler::GetMaxAtbGauge());
+				break;
+		}
+	};
+
+	auto executeOperation = [getAtbValue, operand_flags](int flags, Game_Battler* battler) {
+		switch (flags) {
+			case 0:
+				// set
+				battler->SetAtbGauge(getAtbValue(operand_flags));
+				break;
+			case 1:
+				// add
+				battler->IncrementAtbGauge(getAtbValue(operand_flags));
+				break;
+			case 2:
+				// sub
+				battler->IncrementAtbGauge(-getAtbValue(operand_flags));
+				break;
+		}
+	};
+
+	switch (target_flags) {
+		case 0:
+			// actor
+			if (target_id > 0) {
+				executeOperation(operation_flags, Main_Data::game_actors->GetActor(target_id));
+			}
+			break;
+		case 1:
+			// party member
+			executeOperation(operation_flags, Main_Data::game_party->GetActor(target_id));
+			break;
+		case 2:
+			// entire party
+			for (Game_Battler* member : Main_Data::game_party->GetActors()) {
+				executeOperation(operation_flags, member);
+			}
+			break;
+		case 3:
+			// troop member
+			if (target_id > 0) {
+				executeOperation(operation_flags, Main_Data::game_enemyparty->GetEnemy(target_id));
```

Comment:
``GetActor`` and ``GetEnemy`` return a ``nullptr`` when the target is missing (crashes).

Can you add sanity checks?

```cpp
auto* actor = Main_Data::game_actors->GetActor(target_id); // <- again for game_actors
if (!actor) {
	Output::Warning("CommandManiacControlAtbGauge: Invalid actor ID {}", target_id);
	return true;
}

auto* enemy = Main_Data::game_enemyparty->GetEnemy(target_id);
if (!enemy) {
	Output::Warning("CommandManiacControlAtbGauge: Invalid enemy ID {}", target_id);
	return true;
}
```

Reply:
Thanks for this and great catch! I took your approach as a start, then encapsulated it into a function for reuse, since I have a few calls in this pull request and it will set me up to use in my GetBattleInfo request that I'll post as well.